bitcoin/BRBIP38Key.c

```diff
@@ -181,7 +181,7 @@ int BRKeySetBIP38Key(BRKey *key, const char *bip38Key, const char *passphrase)
     
     BRKeySetSecret(key, &secret, flag & BIP38_COMPRESSED_FLAG);
     var_clean(&secret);
-    BRKeyLegacyAddr(key, address.s, sizeof(address), BIP38_ADDR_PARAMS);
+    BRKeyLegacyAddr(key, address.s, sizeof(address), params);
```

Comment:
Aaron's definition of BIP38_ADDR_PARAMS is:
```
#define BIP38_ADDR_PARAMS      ((BRAddressParams) { BITCOIN_PUBKEY_PREFIX, 0, 0, NULL })
```
where only the first field is in common with `BITCOIN_ADDR_PARAMS`.  It seems only the first field is ever used in BIP38 contexts - do we know that for sure?  Is it known that BIP38 applies to testnet as well?  Given that only the first field is used and the mainnet/testnet differ by the first field, I'd expect BIP38 to apply to testnet as well (even though there is no reason to secure a testnet wallet).

Reply:
We could remove `#define BIP38_ADDR_PARAMS ...` too?

Reply:
The BIP (https://github.com/bitcoin/bips/blob/master/bip-0038.mediawiki) makes no mention of mainnet/testnet. Other implementations support both so its not clear why we would restrict to mainnet.

Yes, BIP38_ADDR_PARAMS can and should be removed. I will amend.